libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKOAuth2.h

```diff
@@ -0,0 +1,116 @@
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import <Foundation/Foundation.h>
+/** SFOAuth default network timeout in seconds.
+ */
+extern const NSTimeInterval kSFOAuthDefaultTimeout;
+
+/** This constant defines the SFOAuth framework error domain.
+ 
+ Domain indicating an error occurred during OAuth authentication.
+ */
+extern NSString * _Nonnull const kSFOAuthErrorDomain;
+
+/**
+ @enum SFOAuthErrorDomain related error codes
+ Constants used by SFOAuthCoordinator to indicate errors in the SFOAuth domain
+ */
+enum {
+    kSFOAuthErrorUnknown = 666,
+    kSFOAuthErrorTimeout,
+    kSFOAuthErrorMalformed,
+    kSFOAuthErrorAccessDenied,              // end user denied authorization
+    kSFOAuthErrorInvalidClientId,
+    kSFOAuthErrorInvalidClientCredentials,  // client secret invalid
+    kSFOAuthErrorInvalidGrant,              // expired access/refresh token, or IP restricted, or invalid login hours
+    kSFOAuthErrorInvalidRequest,
+    kSFOAuthErrorInactiveUser,
+    kSFOAuthErrorInactiveOrg,
+    kSFOAuthErrorRateLimitExceeded,
+    kSFOAuthErrorUnsupportedResponseType,
+    kSFOAuthErrorWrongVersion,              // credentials do not match current Connected App version in the org
+    kSFOAuthErrorBrowserLaunchFailed,
+    kSFOAuthErrorUnknownAdvancedAuthConfig,
+    kSFOAuthErrorInvalidMDMConfiguration,
+    kSFOAuthErrorJWTInvalidGrant
+};
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SFSDKOAuthTokenEndpointErrorResponse : NSObject
+@property  (nonatomic, readonly) NSString *tokenEndpointErrorCode;
+@property  (nonatomic, readonly) NSString *tokenEndpointErrorDescription;
+@property  (nonatomic, readonly) NSError *error;
+@end
+
+
+@interface SFSDKOAuthTokenEndpointRequest : NSObject
```

Comment:
I think some of these properties should be `nonnull`.

Reply:
@bhariharan I need to refine the nullability for  the  properties.